neqo-common/src/qlog.rs

```diff
@@ -29,21 +31,53 @@ pub struct NeqoQlogShared {
 }
 
 impl NeqoQlog {
-    /// Create an enabled `NeqoQlog` configuration.
+    /// Create an enabled `NeqoQlog` configuration backed by a file.
     ///
     /// # Errors
     ///
-    /// Will return `qlog::Error` if cannot write to the new log.
-    pub fn enabled(
-        mut streamer: QlogStreamer,
-        qlog_path: impl AsRef<Path>,
+    /// Will return `qlog::Error` if it cannot write to the new file.
+    pub fn enabled_with_file(
+        mut qlog_path: PathBuf,
+        role: Role,
+        title: Option<String>,
+        description: Option<String>,
+        file_prefix: impl Display,
     ) -> Result<Self, qlog::Error> {
+        qlog_path.push(format!("{file_prefix}.sqlog"));
+
+        let file = OpenOptions::new()
+            .write(true)
+            // As a server, the original DCID is chosen by the client. Using
+            // create_new() prevents attackers from overwriting existing logs.
+            .create_new(true)
+            .open(&qlog_path)
+            .map_err(qlog::Error::IoError)?;
+
+        let streamer = QlogStreamer::new(
+            qlog::QLOG_VERSION.to_string(),
+            title,
+            description,
+            None,
+            std::time::Instant::now(),
+            new_trace(role),
+            qlog::events::EventImportance::Base,
+            Box::new(BufWriter::new(file)),
```

Comment:
This is the whole change. Everything else is part of https://github.com/mozilla/neqo/pull/2031.